### DISCUSSION TITLE
I am having issues with setting a border color for a set of cells.

### DISCUSSION THREAD

**Discussion by db1996** [PHPOffice/PhpSpreadsheet#3768] [Q&A] [answered]
here's the scenario.

A dynamic range of columns and rows are being filled. And all individual cells in that range need to have a black outline.
A certain set of cells within that range needs to have a red outline, they are special cases. however no matter what I try the outline for those cells is always black.

Here's the code I am using right now:

`
        $sheet = $objPHPExcel->getActiveSheet();
	$commentKeys = array_keys($comments);

	$borderStyleRed = [
		'borders' => [
			'outline' => [
				'borderStyle' => \PhpOffice\PhpSpreadsheet\Style\Border::BORDER_THIN,
				'color' => ['argb' => 'FFFF0000']
			]
		]
	];

	$borderStyleBlack = [
		'borders' => [
			'outline' => [
				'borderStyle' => \PhpOffice\PhpSpreadsheet\Style\Border::BORDER_THIN,
				'color' => ['argb' => 'FF000000']
			]
		]
	];

	// Apply black border to each individual cell
	for ($row = 1; $row <= $lastRow; $row++) {
		for ($col = 1; $col <= $lastCol; $col++) {
			$cellCoordinate = \PhpOffice\PhpSpreadsheet\Cell\Coordinate::stringFromColumnIndex($col) . $row;
			$sheet->getStyle($cellCoordinate)->applyFromArray($borderStyleBlack);
		}
	}

	// Now, override with red border only for cells with comments
	foreach ($commentKeys as $cellCoordinate) {
		$sheet->getStyle($cellCoordinate)->applyFromArray($borderStyleRed);
	}
`

If I disable the code to set a black outline entirely, the red outlines are visible, otherwise they are not.

I also tried to use if statements to first check if the cellCoordinate is in the commentkeys, and if it is it should do the red border, otherwise the black border. But even then all outlines are black.

**Comment by oleibman** [comment, marked as answer]
I think you are supplying contradictory information to Excel and it is resolving it as best it can. Let's say that you want cell B2 to have red borders. But it shares borders with cells B1, A2, C2, and B3, all of which are supposed to have black borders. Between cell B1's black bottom border and B2's red top border, something's gotta give.
